projects/hslayers/src/components/map/map.service.ts

```diff
@@ -791,23 +793,50 @@ export class HsMapService {
   repopulateLayers(visibilityOverrides: string[], app: string): void {
     const appConfig = this.hsConfig.get(app);
     if (appConfig.box_layers) {
+      let boxLayers: Layer[] = [];
       appConfig.box_layers.forEach((box) => {
-        for (const lyr of box.getLayers().getArray() as Layer<Source>[]) {
-          this.addLayer(
-            lyr,
-            app,
-            DuplicateHandling.IgnoreNew,
-            visibilityOverrides
-          );
-        }
+        boxLayers = boxLayers.concat(
```

Comment:
Here, I'd like Array.map better. Seems more readable for this use case imo

Reply:
Hmm.. I used concat, because each box is a group, and every group layers gets added to existing array of layers. I think map will only add last groups layers to boxLayers array.

Reply:
I see it now. My bad, sorry